pkg/services/accesscontrol/acimpl/service.go

```diff
@@ -106,19 +108,36 @@ func (s *Service) GetUserPermissions(ctx context.Context, user *user.SignedInUse
 	return s.getCachedUserPermissions(ctx, user, options)
 }
 
-func (s *Service) getUserPermissions(ctx context.Context, user *user.SignedInUser, options accesscontrol.Options) ([]accesscontrol.Permission, error) {
+func (s *Service) getUserPermissions(ctx context.Context, user identity.Requester, options accesscontrol.Options) ([]accesscontrol.Permission, error) {
 	permissions := make([]accesscontrol.Permission, 0)
 	for _, builtin := range accesscontrol.GetOrgRoles(user) {
 		if basicRole, ok := s.roles[builtin]; ok {
 			permissions = append(permissions, basicRole.Permissions...)
 		}
 	}
 
+	namespace, identifier := user.GetNamespacedID()
+	if namespace != authn.NamespaceUser &&
+		namespace != authn.NamespaceServiceAccount &&
+		namespace != identity.NamespaceRenderService {
```

Comment:
question: can render service be assigned permissions outside of the basic roles?

Reply:
yes, when it's acting on behalf of a user. We should revise the namespacing/identifier for it
 https://github.com/grafana/grafana/blob/5efc3386d3fcd933a4a450a038b72a23543a9e83/pkg/services/authn/clients/render.go#L46

Reply:
Oh wow, learn something new every day.